kafka/pykafka/consumer.py

```diff
@@ -47,9 +199,63 @@ class OwnedPartition(object):
     Used to keep track of offsets and the internal message queue.
     """
 
-    def __init__(self, partition):
+    def __init__(self, partition, consumer):
         self.partition = partition
+        self.consumer = consumer
         self._messages = Queue()
+        self.last_offset_consumed = 0
+        self.next_offset = 0
+
+        if self.consumer._auto_commit_enable and self.consumer.consumer_group is not None:
+            self.last_offset_consumed = self._fetch_last_known_offset()
 
     def consume(self, timeout=None):
+        """Get a single message from this partition
+        """
+        if self._messages.empty():
+            self._fetch(timeout=timeout)
+
+        try:
+            message = self._messages.get_nowait()
+            self._queued_chunks -= 1
+            self.last_offset_consumed = message.offset
+            return message
+        except Empty:
+            return None
+
+    def _fetch_last_known_offset(self):
```

Comment:
PS: `last` and `latest` would be confusing as they're the "most recent" in terms of time, but may in fact not be further in the queue than previously committed offsets.  However, as far as the driver is concerned, there is only one committed offset for this consumer group/partition, so there's only one thing to ever be fetched.